### ISSUE TITLE
Esri Daylight widget calendar is inaccessible in Calcite Popover

### ISSUE THREAD

**Issue by Anastasiia-Boleiko** [Esri/calcite-design-system#8033]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

Using Calcite Popover as a container for [Esri Daylight widget](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Daylight.html) makes the date picker of the widget not accessible from the keyboard. 
Opening calendar does not move the focus into the displayed calendar, so a user cannot change a date. 

It looks like the issue is caused by the focus trap in the Calcite Popover, as the Esri calendar widget that is opened from the Daylight widget is placed at the bottom of the DOM, and Calcite popover doesn't treat it as if it belongs to the content and does let focus there. 

Worth mentioning, that the Esri Daylight widget works well when it's not in the Calcite Popover.

### Actual Behavior

Using [Esri Daylight widget](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Daylight.html) in the Calcite Popover makes the widget calendar not accessible from keyboard. 

### Expected Behavior

The [Esri Daylight widget](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Daylight.html) is accessible from keyboard in the Calcite Popover. 

### Reproduction Sample

https://codepen.io/Olga-Knyazeva-the-bashful/pen/bGzbVor

### Reproduction Steps

Use keyboard to navigate through the sample.
1. Open popover by pressing Space/Enter.
2. Navigate to the date picker button using Tabs.
3. Open the calendar by pressing Space/Enter.
4. Try changing the date.

### Reproduction Version

1.9.2

### Working W3C Example/Tutorial

https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-trapping.html

### Priority impact

p2 - want for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Urban

**Comment by jcfranco** [comment]
I believe this is happening because the popover used in the Maps SDK for JS's date picker attaches the itself to the body, so `focus-trap` has no way to know it should be included within the trap (focus strapping is managed within `calcite-popover`'s container). Some ideas:

1. update the Maps SDK for JS popover to use `floating-ui` vs attaching to the body allow its focus to be managed within `calcite-popover`. This one wouldn't require any changes from our side and shouldn't be too complex to tackle.
2. update `calcite-popover` and focus-trapping components to provide a prop to configure `focus-trap` for advanced use cases like this. This would require exploration and would introduce a new config prop on components as a pattern.

Worth noting that if 1 ends up being the immediate solution, it would still be worth it to explore 2 in the future.

cc @driskull

**Comment by driskull** [comment]
Agreed with your assessment.

**Comment by Anastasiia-Boleiko** [comment]
Thanks a lot for the suggestions👍 The solution is found and works well in our case. 
We can close this issue😊
